### PR TITLE
[FEATURE] Extend connection options for clients

### PR DESCRIPTION
**Pull request by josefglatz**
This patch extends the client with three properties:
- host_header: add an optional "Host: yourdomain.com" header
- ignore_cert_errors: force ignoring certificate errors
- force_ip_resolve: force IPv4/-6 resolving of the client

In order to make this possible, the TYPO3 core wrapper
RequestFactory::class for guzzle is used instead of
GeneralUtility::getUrl().

In addition, the following bugs are fixed:

- empty update_comment field on client update
- handle missing disk_total_space (fallback to 0)
- handle missing disk_free_space (fallback to 0)
- handle extensions with not allowed "state" values (fallback to 999 => n/a)
- handle extensions without dependencies

Fixes: #159
Fixes: #160
Fixes: #161
Fixes: #162
Fixes: #163

### CODE REVIEW

File: ext_tables.sql [hunk 1]
```diff
@@ -10,6 +10,9 @@ CREATE TABLE tx_t3monitoring_domain_model_client (
 	domain varchar(255) DEFAULT '' NOT NULL,
 	basic_auth_username varchar(255) DEFAULT '' NOT NULL,
 	basic_auth_password varchar(255) DEFAULT '' NOT NULL,
+	host_header varchar(255) DEFAULT '' NOT NULL,
+    force_ip_resolve enum('', 'v4', 'v6') DEFAULT 'v4' NOT NULL,
```
Comment (georgringer) [inline ext_tables.sql]: spacing

File: README.rst [hunk 2]
```diff
@@ -87,3 +91,7 @@ Especially the import of extensions can take a while, therefore you can use diff
 - ``./typo3/cli_dispatch.phpsh extbase monitoring:importClients`` to fetch the client data
 
 
+Contribution
```
Comment (georgringer) [inline README.rst]: we don't do that, I can write something in release notes as well, pls remove

File: Configuration/TCA/tx_t3monitoring_domain_model_client.php [hunk 3]
```diff
@@ -90,6 +91,40 @@
                 'eval' => 'trim,password'
             ],
         ],
+        'host_header' => [
+            'exclude' => true,
```
Comment (georgringer) [inline Configuration/TCA/tx_t3monitoring_domain_model_client.php]: I guess we could remove all exclude stuff, makes no sense

File: README.rst (hunk 2 above)
Comment (josefglatz) [reply README.rst]: :-[

File: Configuration/TCA/tx_t3monitoring_domain_model_client.php (hunk 3 above)
Comment (josefglatz) [reply Configuration/TCA/tx_t3monitoring_domain_model_client.php]: I would leave it. We have non-admin users. or do you mean this is not evaluated in the module of the extension?

**Comment by josefglatz** [comment]
@georgringer jfyi: I test this patch now in production 🤟

**Pull request merged**